components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -0,0 +1,369 @@
+
+	-- process match
+	Table.mergeInto(match, MatchGroupInput.readDate(match.date))
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+---@param map table
+---@return table
+function CustomMatchGroupInput.processMap(map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+
+	return map
+end
+
+---@param record table
+---@param timestamp integer
+function CustomMatchGroupInput.processOpponent(record, timestamp)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	-- Convert byes to literals
+	if Opponent.isBye(opponent) then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
+
+	---@type number|string
+	local teamTemplateDate = timestamp
+	-- If date is default date, resolve using tournament dates instead
+	-- default date indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not the default date
+	if teamTemplateDate == DateExt.defaultTimestamp then
+		teamTemplateDate = Variables.varDefaultMulti('tournament_enddate', 'tournament_startdate', NOW)
+	end
+
+	Opponent.resolve(opponent, teamTemplateDate, {syncPlayer = true})
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+end
+
+-- called from Module:Match/Subobjects
+---@param player table
+---@return table
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+---@param data table
+---@param indexedScores table[]
+---@return table
+---@return table[]
+function CustomMatchGroupInput.getResultTypeAndWinner(data, indexedScores)
+	-- Map or Match wasn't played, set not played
+	if
+		Table.includes(NP_STATUSES, data.finished) or
+		Table.includes(NP_STATUSES, data.winner)
+	then
+		data.resulttype = 'np'
+		data.finished = true
+	-- Map or Match is marked as finished.
+	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
+	elseif Logic.readBool(data.finished) then
+		if MatchGroupInput.isDraw(indexedScores) then
+			data.winner = 0
+			data.resulttype = 'draw'
+			indexedScores = MatchGroupInput.setPlacement(indexedScores, data.winner, 'draw')
+		elseif MatchGroupInput.hasSpecialStatus(indexedScores) then
+			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
+			data.resulttype = 'default'
+			if MatchGroupInput.hasForfeit(indexedScores) then
+				data.walkover = 'ff'
+			elseif MatchGroupInput.hasDisqualified(indexedScores) then
+				data.walkover = 'dq'
+			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
+				data.walkover = 'l'
+			end
+			indexedScores = MatchGroupInput.setPlacement(indexedScores, data.winner, 'default')
+		else
+			local winner
+			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
```

Comment:
```suggestion
			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
```
```suggestion
			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, nil, data.finished)
```
```suggestion
			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, 1, 2)
```

Reply:
Though this function doesn't return "winner", so you'd need to update that handling